pyuvdata/uvdata/uvdata.py

```diff
@@ -6901,6 +6960,7 @@ def read(
                     read_data=read_data,
                     check_extra=check_extra,
                     run_check_acceptability=run_check_acceptability,
+                    background_lsts=background_lsts,
                 )
 
             elif file_type == "fhd":
```

Comment:
Looks like you didn't add the background_lsts for FHD. Was that on purpose?

Reply:
I need to look closer at the FHD read to make sure the lst calculation can be started further up. Currently it is done at the end and I just have not taken the time yet.

Reply:
That's right, this is why I did not add it. The check on the FHD read to ensure the telescope location is set properly:
```
        try:
            self.set_telescope_params()
        except ValueError as ve:
            warnings.warn(str(ve))

        # need to make sure telescope location is defined properly before this call
        self.set_lsts_from_time_array()
```

Reply:
we can probably move that up too, I can take a look.